### PR TITLE
chore(monorepo): check for misaligned dependencies

### PR DESCRIPTION
**Pull request by mcasimir**
Adds a script for the local env and CI to check for misaligned dependencies:

- errors if NPM would not be able to dedupe and get a single version out of all the version ranges. 
- outputs a warning if finds multiple ranges for the same dependency but that can be deduped.
- allows to add exceptions to a `,depalignrc.json` file to ignore some of the ranges.

### CODE REVIEW

File: scripts/depalign.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+const USAGE = `Check for dependency alignment issues.
+
+USAGE: depalign.js [--skip-deduped|--json|--fix]
+
+Options:
+
+--skip-deduped  don't output warnings for ranges that can be resolve to a single version.
+--json          output a json report
+--suggest-fixes output a list of replacements to normalize ranges and align everything
```
Comment (gribnoysup) [inline scripts/depalign.js:17]: Ahh, even suggestions, that's great! Next step `--autofix` flag 😉

File: scripts/depalign.js [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+          return false;
+        }
+      } catch (e) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
```
Comment (gribnoysup) [inline scripts/depalign.js:80]: On one hand my eye got caught on the amount of unnecessary iterations here due to the nested loops, on the other slicing array inside the loop is probably adding the same amount of overhead, so you decide if you want to apply the suggestion 😆  

```suggestion
function intersects(range) {
  for (const [idx, v1] of Object.entries(range)) {
    for (const v2 of range.slice(Number(idx) + 1)) {
      try {
        if (!semver.intersects(v1, v2)) {
          return false;
        }
      } catch (e) {
        return false;
      }
    }
  }
  return true;
}
```

**Review by gribnoysup — APPROVED** [review]
Looks great!

File: scripts/depalign.js (hunk 2 above)
Comment (mcasimir) [reply scripts/depalign.js:80]: Yeah nice, of course i'll commit it :)! Was O(n^2) for small numbers but it did check the same intersection twice. This looks great, I like the `Object.entries(range)` trick!

**Pull request merged**